docs/blueprints/EP031.rst

```diff
@@ -0,0 +1,273 @@
+
+To illustrate the challenge, consider an EVC terminating on INT Hop Z on port 23. The user packet with INT metadata comes from port 11. **mef_eline** would create the following flows (for simplicity, just one direction is presented):
+
+  0. **met_eline** default behavior:
+
+    - match:
+
+      - priority: DEFAULT (32678)
+      - match: <in_port=11,vlan_vid=20>
+
+    - instruction:
+
+      - action [set_queue, pop_vlan, etc.]
+      - action: output to port 23.
+
+To enable INT, first a physical loop has to be deployed. For this example, on INT Hop Z, port 1 is connected to port 2 by a physical fiber patch cord (done during commissioning). Then, the following flows need to be ADDED to the pipeline:
+
+  1. Adding INT metadata:
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (40000)
+        - match: <in_port= **11**,vlan_vid=20,eth_type=0x800,ip_proto=6>  # TCP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **1** (loop)
+
+        - priority: Higher (40000)
+        - match: <in_port= **11**,vlan_vid=20,eth_type=0x800,ip_proto=17>. # UDP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **1** (loop)
+
+  2. Send Report and pop INT data (traffic is coming from port 2 that's the loop with port 1). Only INT data gets into the loop.
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (40000)
+        - match: <in_port= **2**,vlan_vid=20>
+        - instrutions:
+
+          - action: send_report
+          - action go to table 2
+
+
+      - table 2
+
+        - priority: Higher (40000)
+        - match: <in_port= **2**,vlan_vid=20>
+        - instrutions:
+
+          - action: pop_int
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **23** (original port)
+
+
+IV. How to enable INT for EVCs
+==============================
+
+The goal for the **telemetry** app is to enable telemetry for ALL EVCs. However, it must support enabling and disabling telemetry for a single EVC or ALL EVCs. This is the approach:
+
+  1 . The **telemetry** napp will start operating once **mef_eline** is loaded and EVCs and their flows are pushed to the data plane.
+  2. **telemetry** will listen for events *kytos/mef_eline.(redeployed_link_(up|down)|deployed)* and *kytos.mef_eline.created* issued by **mef_eline**.
+  3. For each EVC identified, **telemetry** will
+    1. use EVC's cookie to get all flow entries created by **flow_manager** IF telemetry is not already enabled.
```

Comment:
I was assuming flow_manager would do consistency of flows. Please elaborate what you think telemetry should do.

Reply:
Section XI addresses that the telemetry napp is responsible for the top consistency.

Reply:
Depending on the events that are subscribed we might not need a consistency check, this could rely on lower levels consistencies or guaranteed events from its dependencies. Unless the top consistency really needs to recover from something and trigger an action like what `sdntrace_cp` does then we think weigh the benefits to maintain an entire new consistency (based on our team experience we know how difficult it can be to stabilize certain kinds of consistency checks that can trigger false positives). Since this NApps essentially "follows" what `mef_eline` does with additional flows, relying on `mef_eline` CRUD events event might suffice especially since `flow_manager` is already ensuring flows consistency. I think it might be worth planning and exploring it without a new consistency, and only add a new one if the existing ones can't provide what  this NApp needs to simplify code maintenance and overall stability.